src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,842 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+//go:build linux
+// +build linux
+
+package debug
+
+import (
+	"archive/zip"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/beevik/ntp"
+	"github.com/docker/go-units"
+	"github.com/hashicorp/go-multierror"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/api/admin"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/cli/cmd/common"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/system"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/system/syslog"
+	"gopkg.in/yaml.v2"
+)
+
+// Use the same date specs as journalctl (see `man journalctl`).
+const timeHelpText = `(journalctl date format, e.g. YYYY-MM-DD)`
+
+type step func() error
+
+type stepParams struct {
+	fs      afero.Fs
+	m       *sync.Mutex
+	w       *zip.Writer
+	timeout time.Duration
+}
+
+type limitedWriter struct {
+	w          io.Writer
+	limitBytes int
+	accumBytes int
+}
+
+func (l *limitedWriter) Write(p []byte) (int, error) {
+	n, err := l.w.Write(p)
+	if err != nil {
+		return n, err
+	}
+
+	l.accumBytes += n
+
+	if l.accumBytes >= l.limitBytes {
+		return n, errors.New("output size limit reached")
+
+	}
+	return n, nil
+}
+
+// Creates a file in the zip writer with name 'filename' and writes 'contents' to it.
+func writeFileToZip(ps *stepParams, filename string, contents []byte) error {
+	ps.m.Lock()
+	defer ps.m.Unlock()
+
+	wr, err := ps.w.Create(filename)
+	if err != nil {
+		return err
+	}
+	_, err = wr.Write(contents)
+	if err != nil {
+		return fmt.Errorf("couldn't save '%s': %w", filename, err)
+	}
+	return nil
+}
+
+// Runs a command and pipes its output to a new file in the zip writer.
+func writeCommandOutputToZipLimit(
+	ps *stepParams,
+	filename string,
+	outputLimitBytes int,
+	command string,
+	args ...string,
+) error {
+	ps.m.Lock()
+	defer ps.m.Unlock()
+
+	ctx, _ := context.WithTimeout(context.Background(), ps.timeout)
```

Comment:
Always worth keeping the `cancel` so as to `defer cancel()`, it's not too critical here, but it frees up timers in Go's runtime timer heap.

Reply:
Fixed.